account_easy_reconcile/easy_reconcile_history.py

```diff
@@ -0,0 +1,145 @@
+    """ Store an history of the runs per profile
+    Each history stores the list of reconciliations done"""
+
+    _name = 'easy.reconcile.history'
+    _rec_name = 'easy_reconcile_id'
+    _order = 'date DESC'
+
+    @api.one
+    @api.depends('reconcile_ids', 'reconcile_partial_ids')
+    def _reconcile_line_ids(self):
+        move_line_ids = []
+        for reconcile in self.reconcile_ids:
+            move_lines = reconcile.mapped('line_id')
+            move_line_ids.extend(move_lines.ids)
+        self.reconcile_line_ids = move_line_ids
+
+        move_line_ids2 = []
+        for reconcile2 in self.reconcile_partial_ids:
+            move_lines2 = reconcile2.mapped('line_partial_ids')
+            move_line_ids2.extend(move_lines2.ids)
+        self.partial_line_ids = move_line_ids2
+
+    easy_reconcile_id = fields.Many2one(
+        'account.easy.reconcile',
+        string='Reconcile Profile',
+        readonly=True
+    )
+    date = fields.Datetime(string='Run date', readonly=True, required=True)
+    reconcile_ids = fields.Many2many(
+        comodel_name='account.move.reconcile',
+        relation='account_move_reconcile_history_rel',
+        string='Partial Reconciliations',
+        readonly=True
+    )
+    reconcile_partial_ids = fields.Many2many(
+        comodel_name='account.move.reconcile',
+        relation='account_move_reconcile_history_partial_rel',
+        string='Partial Reconciliations',
+        readonly=True
+    )
+    reconcile_line_ids = fields.Many2many(
+        comodel_name='account.move.line',
+        relation='account_move_line_history_rel',
+        string='Reconciled Items',
+        _compute='_reconcile_line_ids'
+    )
+    partial_line_ids = fields.Many2many(
+        comodel_name='account.move.line',
+        relation='account_move_line_history_partial_rel',
+        string='Partially Reconciled Items',
+        _compute='_reconcile_line_ids'
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        string='Company',
+        store=True,
+        readonly=True,
+        related='easy_reconcile_id.company_id'
+    )
+
+    @api.multi
+    def _open_move_lines(self, rec_type='full'):
+        """ For an history record, open the view of move line with
+        the reconciled or partially reconciled move lines
+
+        :param history_id: id of the history
+        :param rec_type: 'full' or 'partial'
+        :return: action to open the move lines
+        """
+        assert rec_type in ('full', 'partial'), \
+            "rec_type must be 'full' or 'partial'"
+        move_line_ids = []
+        if rec_type == 'full':
+            move_line_ids = []
+            for reconcile in self.reconcile_ids:
+                move_lines = reconcile.mapped('line_id')
+                move_line_ids.extend(move_lines.ids)
+            name = _('Reconciliations')
+        else:
+            move_line_ids = []
+            for reconcile in self.reconcile_partial_ids:
```

Comment:
This for loop can be replaced by:

`move_line_ids = self.mapped('reconcile_ids.line_partial_ids').ids`